kube-timer-instruction.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Instruction for kube-timer
+description: Learn how to use kube-timer to calculate the provision/deletion time of Kubernetes resources.
+author: CocoWang-wql
+date: 07/3/2022
+---
+
+# Use kube-timer to get the provision/deletion time of kubernetes resources
+
+This articles provides the instructions about how get the provision/deletion time of kubernetes resources. It contains the installtion steps, description of parameters and samples outputs.
+
+## How to install kube-timer
+
+You could run the below command in the terminal to install kube-timer tool.
+
+```shell
+go install github.com/nilo19/kube-timer
+```
+
+After the installtion, run `kube-timer -h` to check it.
+
+## Parameters of kube-timer
+| Parameter | Description |
+| --------- | ------------- |
+| -f | The yaml file for resources provision | 
+| -c | The number of resource | 
+| -a | Create all the resources together and wait for the previous resouces to be ready. The time of single resource creation would contain waiting in queue time. Without this parameter, the resources are created one by one; the next resource will only be created once the previous one is ready.|
+| -d | Delete the resource |
+| -n | Name of the resource to be leleted |
+| -D | Delete all the resources together | 
+| --started-event-reason | The reason of event. It can be used as the start time of provision/deletion |
+| --finished-event-reason | The reason of event. It can be used as the end time of provision/deletion |
+| --namespace | The namespace of resources | 
+
+## Samples for the kube-timer
+
+### Get the provision time of single LoadBalancer service.
+1. Create a file named `svc.yaml` 
+
+2. Sample YAML definition:
+
+```yml
+kind: Service
+apiVersion: v1
+metadata:
+  Name: testsvc
+  namespace: default
+spec:
+  selector:
+    app: nsg-test
+  ports:
+    - name: port1
+      port: 81
+      targetPort: 81
+      protocol: TCP
+  type: LoadBalancer
+  
+```
+
+3. Get the provision time of this LoadBalancer service.
+```shell
+kube-timer svc -f <service definition yaml file>
+```
+
+### Get the provision time of multiple LoadBalancer services.
+
+#### Do not contain waiting-in-queue time
+
+As we do not specify `-a`, all the services will be created one by one. The next service is only be created once the previous one is ready.
```

Comment:
This is the behavior of the cloud provider azure, we shouldn't assume all providers work like that.